test/scope_visitor_test.dart

```diff
@@ -0,0 +1,184 @@
+@TestOn('vm')
+import 'package:analyzer/dart/analysis/utilities.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:dart_code_metrics/src/models/class_type.dart';
+import 'package:dart_code_metrics/src/models/function_type.dart';
+import 'package:dart_code_metrics/src/scope_visitor.dart';
+import 'package:path/path.dart' as p;
+import 'package:test/test.dart';
+
+void main() {
+  group('ScopeVisitor collects scope from file with', () {
+    ScopeVisitor visitor;
+
+    setUp(() {
+      visitor = ScopeVisitor();
+    });
+
+    test('abstract class', () async {
+      (await resolveFile(
+        path: p.normalize(p.absolute('./test/resources/abstract_class.dart')),
+      ))
+          .unit
+          .visitChildren(visitor);
+
+      final classDeclaration = visitor.classes.single;
+      expect(classDeclaration.declaration, isA<ClassDeclaration>());
+      expect(classDeclaration.type, equals(ClassType.generic));
+      expect(classDeclaration.name, equals('Doer'));
+
+      final function = visitor.functions.single;
+      expect(function.type, equals(FunctionType.method));
+      expect(function.name, equals('doSomething'));
+      expect(function.fullName, equals('Doer.doSomething'));
+    });
+
+    test('class with factory constructors', () async {
+      (await resolveFile(
+        path: p.normalize(p
+            .absolute('./test/resources/class_with_factory_constructors.dart')),
```

Comment:
Suggestion: what do you think about moving path declaration to a separate variable (same for all test cases)?